test/db/migration-golang-migrate/mysql_test.go

```diff
@@ -34,32 +38,38 @@ func (s *MysqlTestSuite) SetupSuite() []suite.Option {
 }
 
 func (s *MysqlTestSuite) TestPlainFixturesMysql() {
-	ctx := s.Env().Context()
+	envContext := s.Env().Context()
 	envConfig := s.Env().Config()
 	envLogger := s.Env().Logger()
 	envClient := s.Env().MySql("default").Client()
 
-	loader := fixtures.NewFixtureLoader(ctx, envConfig, envLogger)
-	err := loader.Load(ctx, plainMysqlTestFixtures())
+	loader := fixtures.NewFixtureLoader(envContext, envConfig, envLogger)
+
+	dbFixtures, writer, enabled, purge, err := provideFixtureData(envContext, envConfig, envLogger)
+	err = loader.Load(envContext, []fixtures.FixtureSet{fixtures.NewSimpleFixtureSet(dbFixtures, writer, enabled, purge)})
 	s.NoError(err)
 
 	gosoAssert.SqlTableHasOneRowOnly(s.T(), envClient, "mysql_plain_writer_test")
 	gosoAssert.SqlColumnHasSpecificValue(s.T(), envClient, "mysql_plain_writer_test", "name", "testName3")
 }
 
-func plainMysqlTestFixtures() []*fixtures.FixtureSet {
-	return []*fixtures.FixtureSet{
-		{
-			Enabled: true,
-			Purge:   false,
-			Writer: fixtures.MysqlPlainFixtureWriterFactory(&fixtures.MysqlPlainMetaData{
-				TableName: "mysql_plain_writer_test",
-				Columns:   []string{"id", "name"},
-			}),
-			Fixtures: []interface{}{
-				fixtures.MysqlPlainFixtureValues{2, "testName2"},
-				fixtures.MysqlPlainFixtureValues{2, "testName3"},
-			},
-		},
+func provideFixtureData(ctx context.Context, config cfg.Config, logger log.Logger) (fixtures.NamedFixtures[fixtures.MysqlPlainFixtureValues], fixtures.FixtureWriter, bool, bool, error) {
```

Comment:
can't this one directly return the FixtureSet?

Reply:
fixed